src/modules/file/file.usecase.ts

```diff
@@ -127,6 +129,50 @@ export class FileUseCases {
     return newFile;
   }
 
+  async updateFileMetaData(
+    user: User,
+    fileUuid: File['uuid'],
+    newFileMetada: UpdateFileMetaDto,
+  ) {
+    const file = await this.fileRepository.findOneBy({
+      uuid: fileUuid,
+      status: FileStatus.EXISTS,
+    });
+
+    if (!file.isOwnedBy(user)) {
+      throw new ForbiddenException('This file is not yours');
+    }
+
+    const cryptoFileName = this.cryptoService.encryptName(
+      newFileMetada.plainName,
+      file.folderId,
+    );
+
+    const fileWithSameNameExists = await this.fileRepository.findOneBy({
+      name: cryptoFileName,
```

Comment:
Search by plainName also (using OR) as we are going to migrate the field name, so we need to prevent a duplication on the plainName field also